src/service/resources-impl.js

```diff
@@ -866,12 +869,23 @@ export class Resources {
         const resource = request.resource;
         const box = resource.getLayoutBox();
         const iniBox = resource.getInitialLayoutBox();
-        const diff = request.newHeight - box.height;
+        const margins = resource.getLayoutMargins();
```

Comment:
I don't think just doing what you suggest is correct for either the case of changing padding or margins.

It seems that the existing logic uses the fact that the top edge of an element will not change as a result of a change in height (none of the rules for setting resize=true/false ever check what box.top is) and for changes to the height where margins and padding are held constant this is valid.

As far as a user is concerned the top edge of an element is the edge of whatever they can see. For the case of an image without a border, then this will be the top of the content box, for the case of something with a border or a background color this will be the border box (which is equivalent to layout box in AMP).

So in all cases, changing the top margin is more akin to re-positioning the element than it is to resizing it. And for elements with no border or background changing the top padding would also be more akin to re-positioning than resizing.

So changes in margins need to be considered somewhat separately from changes in height if we want to avoid user visible re-flows in the middle of the viewpoint.